lib/node_modules/@stdlib/random/base/mt19937/lib/factory.js

```diff
@@ -0,0 +1,163 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var setReadOnly = require('@stdlib/utils/define-read-only-property');
+var isPositiveInteger = require('@stdlib/assert/is-positive-integer').isPrimitive;
+var INT32_MAX = require('@stdlib/constants/math/int32-max');
+var UINT32_MAX = require('@stdlib/constants/math/uint32-max');
+var randint32 = require('./rand_int32.js');
+
+
+// VARIABLES //
+
+var NORMALIZATION_CONSTANT = (UINT32_MAX) | 0; // asm type annotation
+var MAX_SEED = (INT32_MAX - 1) | 0; // asm type annotation
+var N = 624;
+var M = 397;
+var UPPER_MASK = 0x80000000; /* Most significant w-r bits */
+var LOWER_MASK = 0x7fffffff; /* Least significant r bits */
+var MATRIX_A = 0x9908b0df;
+
+
+// MAIN //
+
+/**
+* Returns a 32-bit Mersenne Twister pseudorandom number generator.
+*
+* @param {PositiveInteger} [seed] - pseudorandom number generator seed
+* @throws {TypeError} must provide a positive integer
+* @throws {RangeError} must provide a positive integer less than the maximum signed 32-bit integer
+* @returns {Function} Mersenne Twister PRNG
+*
+* @example
+* var mt19937 = factory();
+*
+* var v = mt19937();
+* // returns <number>
+*
+* @example
+* // Return a seeded Mersenne Twister PRNG:
+* var mt19937 = factory( 1234 );
+*
+* var v = mt19937();
+* // returns 822569775
+*/
+function factory( seed ) {
+	var state;
+	var mti;
+	var mt = new Array( N );
+	var s;
+
+	if ( arguments.length ) {
+		if ( !isPositiveInteger( seed ) ) {
+			throw new TypeError( 'invalid input argument. Must provide a positive integer. Value: `' + seed + '`.' );
```

Comment:
We just recently changed our error message convention.

s/invalid input argument/invalid argument/

Both L72 and L75.